Shorten/improve intro of "Basic proof writing" chapter. (#13651)
coq/coq: doc/sphinx/proofs/writing-proofs/index.rst

tactics which the user will use to represent steps in the proof of a
theorem.

A first section presents the proof mode (the core mechanism of the
dialog between the user and the proof assistant).  Then, several
sections describe the available tactics.  One section is dedicated to
the SSReflect proof language, which proposes a consistent set of
tactics to replace some of the standard basic tactics.  Finally, the
last section documents the ``Scheme`` family of commands, which can be
used to extend the power of the :tacn:`induction` and

> :tacn:`inversion` tactics.

Reviewer: This would probably be more readable if presented as a bullet list.  Then it wouldn't be necessary to repeat "section" so may times.  The bullet points could include links to the described sections.

Author: The bullet point list of sections is just below though.

Reviewer: Good point--I only looked at what's in GitHub.  Then I'd suggest adding a little bit of description for each bullet point rather than in a separate paragraph that covers all of them.

Author: Unfortunately, this list is auto-generated by Sphinx. I don't know how it can be customized. But another reason for not adding a description for each section just yet is that this chapter is not yet in its final form (the tactics chapter and the detailed examples chapter should go away in favor of multiple new sections). But this will unfortunately have to wait for 8.14.

Reviewer: Looks like the `:hidden:` or `:includehidden:` options on `toctree` would let you hide the auto-generated list.  OTOH, if we then provide our own list, we have to remember to update it when we add/remove/rename rst files.

https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html

Up to you whether you want to try that now or leave it for another day.

Author: Let's leave it for later. I think it will make sense to refine the presentation of the sections when the list of sections is final.

> A first section presents the proof mode (the core mechanism of the
> dialog between the user and the proof assistant).  Then, several
> sections describe the available tactics.  One section is dedicated to
> the SSReflect proof language, which proposes a consistent set of
> tactics to replace some of the standard basic tactics.  Finally, the
> last section documents the ``Scheme`` family of commands, which can be
> used to extend the power of the :tacn:`induction` and
> :tacn:`inversion` tactics.

Reviewer: ```suggestion
The first section presents the proof mode (the core mechanism of the
dialog between the user and the proof assistant).  Then, several
sections describe the available tactics.  One section covers
the SSReflect proof language, which proposes a consistent set of
tactics to replace some of the standard basic tactics.  The
last section documents the ``Scheme`` family of commands, which can be
used to extend the power of the :tacn:`induction` and
:tacn:`inversion` tactics.
```

For SSR, instead of "which proposes a consistent set of
tactics to replace some of the standard basic tactics", how about
"provides a consistent alternative set of tactics to the standard basic tactics"?

> Additional tactics will be documented in the next chapter
> :ref:`automatic-tactics`.

Reviewer: ```suggestion
Additional tactics are documented in the next chapter
:ref:`automatic-tactics`.
```
